README.md

```diff
@@ -453,6 +457,108 @@ func FunctionReconciler(c reconcilers.Config) *reconcilers.ResourceReconciler[*b
 ```
 [full source](https://github.com/projectriff/system/blob/4c3b75327bf99cc37b57ba14df4c65d21dc79d28/pkg/controllers/build/function_reconciler.go#L39-L51)
 
+#### IfThen
+
+An [`IfThen`](https://pkg.go.dev/github.com/vmware-labs/reconciler-runtime/reconcilers#IfThen) branches execution of the current reconcile request based on a condition. The false `Else` branch is optional and ignored if not defined.
+
+**Example:**
+
+An IfThen can be used to gate a capability of the reconciler only for a resource that opts-in to the behavior. 
+
+```go
+func GatedReconciler() *reconcilers.SubReconciler[*buildv1alpha1.Function] {
+	return &reconcilers.IfThen[*buildv1alpha1.Function]{
+		If: func(ctx context.Context, resource *buildv1alpha1.Function) bool {
+			return resource.Labels["capability-gate.example/enabled"] == "true"
+		}
+		Then: reconcilers.Sequence[*buildv1alpha1.Function]{
+			// use the gated feature
+		},
+	}
+}
+```
+
+#### While
+
+A [`While`](https://pkg.go.dev/github.com/vmware-labs/reconciler-runtime/reconcilers#While) calls the reconciler so long as the condition is true, up to the maximum number of iterations (defaults to 100). The current iteration index can be retried with [`RetrieveIteration`](https://pkg.go.dev/github.com/vmware-labs/reconciler-runtime/reconcilers#RetrieveIteration).
+
+This reconciler must not be used to wait for an external to change, or for polling as this will block the reconciler queue. It to return with the result requesting to be requeued, or to watch the external state for changes that enqueue the reconcile request.
```

Comment:
```suggestion
This reconciler must not be used to wait for external state to change, or for polling as this will block the reconciler queue. Its use is to return with the result requesting to be requeued, or to watch the external state for changes that enqueue the reconcile request.
```
Not sure whether I understand the second sentence.